From 1fab9b9d7b05a9b40084712a5a60fee49bf554b1 Mon Sep 17 00:00:00 2001
From: Roger Shimizu <rogershimizu@gmail.com>
Date: Sat, 17 Aug 2019 01:22:36 +0900
Subject: [PATCH] Fix FTBFS under GCC-9

Resolve report from Debian:
 - https://bugs.debian.org/925829

obfs_http.c: In function 'check_http_header':
obfs_http.c:247:13: error: 'strncasecmp' specified bound
18446744073709551614 exceeds maximum object size 9223372036854775807
[-Werror=stringop-overflow=]
  247 |         if (strncasecmp(hostname, obfs_http->host, result) == 0)
{
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
---
 src/obfs_http.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/obfs_http.c b/src/obfs_http.c
index 9051222..30eae6b 100644
--- a/src/obfs_http.c
+++ b/src/obfs_http.c
@@ -245,7 +245,7 @@ check_http_header(buffer_t *buf)
             }
 
         result = OBFS_ERROR;
-        if (strncasecmp(hostname, obfs_http->host, result) == 0) {
+        if (strncasecmp(hostname, obfs_http->host, len) == 0) {
             result = OBFS_OK;
         }
         free(hostname);
